src/goTest.ts

```diff
@@ -116,19 +116,23 @@ export function testCurrentFile(timeout: string) {
  *
  * @param config the test execution configuration.
  */
-function goTest(config: TestConfig): Thenable<boolean> {
+function goTest(config: TestConfig, goConfig?: vscode.WorkspaceConfiguration): Thenable<boolean> {
 	return new Promise<boolean>((resolve, reject) => {
 		outputChannel.clear();
 		outputChannel.show(2);
-		let buildFlags: string[] = vscode.workspace.getConfiguration('go')['buildFlags'];
-		let buildTags: string = vscode.workspace.getConfiguration('go')['buildTags'];
+		if (!goConfig) {
+			goConfig = vscode.workspace.getConfiguration('go');
+		}
+		let buildFlags: string[] = goConfig['buildFlags'];
+		let buildTags: string = goConfig['buildTags'];
 		let args = ['test', '-v', '-timeout', config.timeout, '-tags', buildTags, ...buildFlags];
+		let testEnvVars = Object.assign({}, goConfig['testEnvVars'], process.env);
```

Comment:
Doesn't the order need to be changed to ensure that the `testEnvVars` overrride the process environment settings?